__tests__/notifications.ts

```diff
@@ -291,6 +302,118 @@ describe('query notifications', () => {
   });
 });
 
+describe('query notificationPreferences', () => {
+  const QUERY = `
+    query NotificationPreferences($data: [NotificationPreferenceInput]!) {
+      notificationPreferences(data: $data) {
+        referenceId
+        userId
+        notificationType
+        status
+        type
+      }
+    }
+  `;
+
+  it('should not authorize when not logged-in', () =>
+    testQueryErrorCode(
+      client,
+      { query: QUERY, variables: { data: [] } },
+      'UNAUTHENTICATED',
+    ));
+
+  const prepareNotificationPreferences = async () => {
```

Comment:
I would suggest making this more extended for future proofing.

For instance:
- add multiple of the same type for different posts but same user
- also have on test that only queries one of the types and other should not auto return


Its just a failsafe for when code gets accidentally changed
Non blocking so will approve already

Reply:
> add multiple of the same type for different posts but same user

Sure. Let me add another one.

> also have on test that only queries one of the types and other should not auto return

We should have this covered as we already have other data for user `1` but with `source` type. First first test's return length was just one and verified to have the correct type (post).

Reply:
Added! Thanks for the review! 🚀